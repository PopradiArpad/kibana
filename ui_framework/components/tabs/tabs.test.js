import React from 'react';
import { render, shallow } from 'enzyme';
import { requiredProps } from '../../test/required_props';
import sinon from 'sinon';

import {
  KuiTabs,
} from './tabs';

import {
  KuiTab,
} from './tab';

describe('KuiTabs', () => {
  const tabs = [
    'Cobalt',
    'Dextrose',
    'Helium-3',
    'Monosodium Glutamate',
  ];

  test('renders', () => {
    const component = (
      <KuiTabs
         selectedTabIndex={0}
         onSelectedTabChanged={()=>{}}
         { ...requiredProps }
       >
        {tabs}
      </KuiTabs>
    );

    expect(render(component)).toMatchSnapshot();
  });

  describe('Props', () => {
    describe('onSelectedTabChanged', () => {
      test(`isn't called upon instantiation`, () => {
        const onSelectedTabChangedHandler = sinon.spy();

        shallow(
          <KuiTabs
             selectedTabIndex={0}
             onSelectedTabChanged={onSelectedTabChangedHandler}
           >
            {tabs}
          </KuiTabs>
        );

        sinon.assert.notCalled(onSelectedTabChangedHandler);
      });

      test('is called when the button is clicked', () => {
        const onSelectedTabChangedHandler = sinon.spy();

        const component = shallow(
          <KuiTabs
             selectedTabIndex={0}
             onSelectedTabChanged={onSelectedTabChangedHandler}
           >
            {tabs}
          </KuiTabs>
        );

        component.find(KuiTab).at(1).simulate('click');
        sinon.assert.calledOnce(onSelectedTabChangedHandler);
        sinon.assert.calledWith(onSelectedTabChangedHandler,1);
      });
    });
  });
});
