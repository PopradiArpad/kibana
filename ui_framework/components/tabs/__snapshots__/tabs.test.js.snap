// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KuiTabs renders 1`] = `
<div
  aria-label="aria-label"
  class="kuiTabs testClass1 testClass2"
  data-test-subj="test subject string"
>
  <button
    class="kuiTab kuiTab-isSelected"
  >
    Cobalt
  </button>
  <button
    class="kuiTab"
  >
    Dextrose
  </button>
  <button
    class="kuiTab"
  >
    Helium-3
  </button>
  <button
    class="kuiTab"
  >
    Monosodium Glutamate
  </button>
</div>
`;
